test/powershell/Modules/Microsoft.PowerShell.Management/Start-Process.Tests.ps1

```diff
@@ -104,6 +104,10 @@ Describe "Start-Process" -Tags @("Feature") {
         $process.Name | Should Be "notepad"
         $process | Stop-Process
     }
+
+    It "Should be able to use the whatif switch without error" {
+        { Start-Process $pingCommand -ArgumentList $pingParam -WhatIf } | Should Not Throw
+    }
```

Comment:
Perhaps add a test that would normally perform an action and ensure that `-WhatIf` prevented the action? 
For example, have a `Start-Process` command to write a string to a file and then test to ensure the file was not written to with `-WhatIf` is supplied.

Reply:
Maybe also a test that using the `PassThru` switch, no object gets returned (which is also a good additional test that the command was not executed)?
```
It "returns null when using -WhatIf switch with `PassThru`" {
     { Start-Process $pingCommand -ArgumentList $pingParam -PassThru -WhatIf } | Should Be $null
}
```

Reply:
Thanks for your suggestion @bergmeister . I've added the test that you described.